### DISCUSSION TITLE
Pyrosim - hydrogen Gas release

### DISCUSSION THREAD

**Discussion by Gokul-dev823** [firemodels/fds#13268] [General]
Dear community,

I'm currently learning and simulating a scenario in FDS using Pyrosim and I'm really stuck in a particular area where I would like some suggestions to move forward. I'm trying to simulate a hydrogen gas release from the stove in the kitchen at a release velocity of 97 m/s by creating an obstruction over the stove and defining the surface. I'm getting results which 'm not expecting and that is the fire seems to be very large in the output with more than 2m and spreading over the ceiling. other Sources and validations says that the fire at this velocity cannot be more than 24cm. I have attahced the code file and would be happy to receive any comments suggestions for me to proceed further. I would also like to know where I'm getting it wrong in the code and how to exactly measure the flame height.  Thanks
[Kitchen_Sim.txt](https://github.com/user-attachments/files/16472353/Kitchen_Sim.txt)

**Comment by mcgratta** [comment]
You should not use `MASS_FRACTION` and `VEL` on the `SURF` line. FDS is applying these to your `THICKENED` obstruction and producing far more mass. Use `MASS_FLUX` and apply to an area comparable to a grid cell. This will produce an equivalent mass flow.

**Reply by Gokul-dev823** [reply]
Thanks for your reply. I will run the simulation now and let you know how I am getting on.

**Reply by Gokul-dev823** [reply]
Im trying to simulate the real case here, so for it I have calculated the velocity of 97.7m/s considering real case of 4mbar pressure from the burner. Now, For the calculated mass flux of 8.8kg/m2s considering the above and assigned it to the obstruction, I could able to see any flame from the output. I ran the simulation for about 60secs. However i defined it under "general surface" and not as a supply.

&SURF ID='Surface02',
      COLOR='RED',
      MASS_FLUX_TOTAL=8.8,
      TMP_FRONT=0.0,
      MLRPUA=0.0/

**Comment by drjfloyd** [comment]
&SURF ID='Surface02',
COLOR='RED',
MASS_FLUX_TOTAL=8.8,
TMP_FRONT=0.0,
MLRPUA=0.0/ 

Where have you told FDS that your mass flux is hydrogen?

**Reply by drjfloyd** [reply]
You are using a CFD code. You have access to basically every variable that FDS computes via outputs.  In the case of not getting a fire you should be thinking about what might cause this?  What are the conditions you need for a fire? You need fuel, you need oxygen, and you need to have those present in conditions that pass the extinction check. All of these are things you could have added as outputs in an attempt to figure out why your inputs are not working.

**Reply by Gokul-dev823** [reply]
&REAC ID='FDS6 HYDROGEN',
      FYI='FDS6 Predefined',
      FUEL='HYDROGEN'/
I defined these predefined reaction for hydrogen. And the mass flux definition under SURF would be taking considering this reaction right?

**Reply by drjfloyd** [reply]
Have you read the User's Guide about MASS_FLUX_TOTAL?

**Reply by Gokul-dev823** [reply]
Yes, i went through it yet I'm finding it hard to understand. between writing a code manually and with pyrosim.

**Reply by drjfloyd** [reply]
From 10.1.2:

Note that MASS_FLUX_TOTAL should only be used for an outflow boundary condition; for inflow use MASS_FLUX which is discussed in Sec. 10.1.6.

**Reply by Gokul-dev823** [reply]
yes i just got an understanding over in it and I get what you mean. now even with specifying hydrogen with mass flux of 8.8, I still get the same result except there is increase in velocity in the release region which I understood from slice. and below is the code I have used.

&SURF ID='Surface02',
      TMP_FRONT=0.0,
      MASS_FLUX=8.8,
      SPEC_ID='HYDROGEN',
      TAU_MF=1.0/
I'm trying to analyze why I'm not able to visualize fire, and I'm sure at these conditions with vents, fuel and the rate the fire should happen. When i try with defining the specific velocity of 94.7m/s the fire is huge or else no fire with either volume flow rate or mass flux at this rate.

**Reply by drjfloyd** [reply]
Did you carefully read the MASS_FLUX section of the Guide?

The mass fraction of the background species will be set to account for any mass fraction not specified with other species.

What mass fraction of hydrogen did you specify?

**Reply by Gokul-dev823** [reply]
i have specified earlier in the code and as 1 like this 
&SPEC ID='HYDROGEN', MASS_FRACTION_0=1.0/

**Reply by drjfloyd** [reply]
Please carefully read the User's Guide description of what MASS_FRACTION_0 on &SPEC does. You need to take the time to make sure you understand what the various FDS inputs mean.
